satellite/contact/client.go

```diff
@@ -0,0 +1,44 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package contact
+
+import (
+	"context"
+
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/transport"
+)
+
+type client struct {
+	log    *zap.Logger
+	conn   *grpc.ClientConn
+	client pb.ContactClient
+}
+
+// newClient dials the target contact endpoint
+func newClient(ctx context.Context, log *zap.Logger, transport transport.Client, target *pb.NodeAddress) (*client, error) {
```

Comment:
It's occurred to me that we could do the node ID check with less code, using existing functionality. This function could take a new argument `expectedNodeID` and then make a new `dialOption` like

```go
    dialOption, err := transport.tlsOpts.DialOption(expectedNodeID)
```

and then pass that dialOption to `transport.DialAddress()` and that would tell grpc to add a certificate verifier function that checks for the remote cert matching an expected nodeID.

I'm not 100% sure it would work, because `DialAddress()` adds a conflicting `grpc.WithTransportCredentials`, but it adds our additional options to the end of the list and I would guess that later options override earlier ones.

Reply:
hm. interesting. let me give this a shot.

Reply:
Alright I added the dialOption, see commit: https://github.com/storj/storj/pull/2952/commits/0ed3cfdea57c5d97c5909cc510872a8684ba6de7

Does it look right?

Reply:
It does! Yes. Looks like you did what I meant, instead of the broken thing that I said :D

Reply:
lol, no no you explained it perfectly